FlowCrypt/Functionality/Mail Provider/Message Provider/ProcessedMessage.swift

```diff
@@ -111,8 +112,13 @@ extension ProcessedMessage {
         self.message = message
         self.type = .plain
         if let html = message.body.html {
-            self.text = try await Core.shared.sanitizeHtml(html: html)
-            self.quote = nil
+            let (text, quote) = Self.parseHtmlQuote(from: html)
+            self.text = try await Core.shared.sanitizeHtml(html: text)
+            if let quote {
+                self.quote = try await Core.shared.sanitizeHtml(html: quote)
+            } else {
+                self.quote = nil
+            }
```

Comment:
SwiftSoup can sanitize HTML too (https://github.com/scinfu/SwiftSoup?tab=readme-ov-file#sanitize-untrusted-html-to-prevent-xss), probably we can use it so sanitizing will be performed natively instead of using JS?

Reply:
Yeah, let's leave it for now, as current implementation works well and it can be quite complicated to implement the same logic using SwiftSoup.

Reply:
Okay